core/store/src/flat/types.rs

```diff
@@ -119,6 +128,29 @@ pub enum FlatStorageCreationStatus {
     CatchingUp(CryptoHash),
 }
 
+/// This struct represents what is the current status of flat storage resharding.
+/// During resharding flat storage must be changed to reflect the new shard layout.
+///
+/// When two shards are split, the parent shard disappears and two children are created. The flat storage
+/// entries that belonged to the parent must be copied in one of the two shards. This operation happens in the
+/// background and could take significant time.
+/// After all elements have been copied the new flat storages will be behind the chain head. To remediate this issue
+/// they will enter a catching up phase. The parent shard, instead, must be removed and cleaned up.
+#[derive(
+    BorshSerialize, BorshDeserialize, Clone, Debug, PartialEq, Eq, serde::Serialize, ProtocolSchema,
+)]
+pub enum FlatStorageReshardingStatus {
```

Comment:
Thinking about the future, the next event that we'll be adding will be merging. Can you make the names of events future proof - e.g. think what will be good names for the merging parents and children and make it so that it's consistent when we add those.